packages/babel-plugin/src/styled/index.tsx

```diff
@@ -32,7 +32,7 @@ const extractStyledDataFromNode = (
     t.isMemberExpression(node.callee) &&
     t.isIdentifier(node.callee.object) &&
     node.callee.object.name === 'styled' &&
-    t.isObjectExpression(node.arguments[0]) &&
+    t.isExpression(node.arguments[0]) &&
```

Comment:
we widen the typecheck so any expression can be passed through (such as arrays, identifiers, and so on)

Reply:
What all we have to support here? `isExpression` will also return true for expression of any type including `ArrowFunctionExpression`. Should we limit this 🤔 ? Do we have to handle nested arrays? Didn't dig deep but something like this:

```typescript
    const ListItem5 = styled.div([
        styles,
        `color: blue;`,
        { fontWeight: 500 },
       [`background-color: red`]
      ]);
```

Reply:
I think we did handle. There is a recursion in `buildCss` 😅

Reply:
We can add tests for nested arrays thought. They will get flattened with recursion.

Reply:
> `isExpression` will also return true for expression of any type including `ArrowFunctionExpression`

ahh!! we are throwing error for unsupported node in `buildCss` 🎉

Reply:
yeah!

Reply:
> Do we have to handle nested arrays?

i think it could work but its not officially supported atm - at least from the perspective of the style typing. maybe for now we leave it?

Reply:
Yeah no worries man. I was just curious to know. Loved the work ❤️